### DISCUSSION TITLE
Unable to find a downloadable video with ZDF (German television)

### DISCUSSION THREAD

**Discussion by Ingo62a** [aclap-dev/video-downloadhelper#1362] [Help and Q&A]
### Your question

I don't know whether the ZDF (German public television) changed any of its broadcasting, or whether the download helper (beta) changed, but some videos are no longer recognized, e.g.

https://www.zdf.de/serien/die-rosenheim-cops/stockls-verdacht-100.html

I've had similar problems with the ARD (another German public television channel) but there switching to "auto" quality and back to a particular one like 1080p "fixed" the detection problem.

With the ZDF some movies still work, while series (like the one linked) are not recognized anymore, suggesting that it might be related to the ZDF changing its display mode ...

### Addon & CoApp version and details

using 9.1.0.41 and 2.0.19

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 24H2 build 26100.2033, Firefox 131.0.3 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.  There's some bits of information there that you omitted.  You really ought to include those when you post here.  But at least you provided a URL, which is usually the main piece of useful information.

![#01](https://github.com/user-attachments/assets/f5077583-64d8-4595-8a6b-a2750ee7650e)
![#02](https://github.com/user-attachments/assets/3002e391-8c15-4895-bb8f-b248295ef89f)
![#03](https://github.com/user-attachments/assets/55cc580b-2bab-40a4-a06c-5d2008f5b251)
![#04](https://github.com/user-attachments/assets/67c19c78-5892-4a2d-a3f0-cbdf3c082240)
![#05](https://github.com/user-attachments/assets/a2850277-3f5b-475c-8243-2b7cada45384)

I looked for captions for this, hoping there might be English subtitles on offer.  But I didn't find any.  I think there are German captions but I couldn't find even those.

I must say I'm somewhat surprised I wasn't geoblocked on this one.  So much content in the EU is geoblocked for me over here in the US.

In any case, this one seemed to pose no problem whatsoever for VDH.  That's not to say that everything will work.  This particular show was not protected by DRM.  There seems to be an epidemic of DRM sweeping the world.  Perhaps content you are having trouble with is protected by DRM.  Firefox puts up a little DRM badge in the same line with the URL at the top of the window, to the left of the URL.  Hover your mouse over those badges.  If one of them mentions DRM, that particular content is protected.  If you see that, forget about downloading.

**Comment by Ingo62a** [comment]
Thanks! This is weird, as it now works for me, too! 

I had tied multiple times without success, this particular show didn't show up when running, while it worked for other videos which I tried to validate my installation in itself wasn't screwed up in the first place.

For now my apologies causing you extra effort :-(

**Comment by Ingo62a** [comment]
Reconsidering, if seems the channel managed to somehow block a download of the episode while still broadcasted live although showing up in their library for streaming already. Only after its live broadcasting ended the episode could be downloaded, too.

At least this appears to be the most likely reason why it didn't work before and does now without any other change or browser restart.

**Reply by GrampaWildWilly** [reply]
That sounds a lot like something I've encountered on YouTube on occasion.  While the item is live, VDH doesn't download it properly.  It's been a while since I've hit this so my memory is a bit hazy on this.  But as I recall, what VDH would download while the thing was live was only part of it.  It might have been the beginning up to the point that had been broadcast at the moment I launched the download.  But if I waited some unknown amount of time after the stream ended, maybe half an hour, maybe longer, then the download was fine.  But there's other sites that work fine to record a livestream while it is live.  Medici TV works fine.  BBC3 Radio won't let you capture anything live but their archive works fine.  I don't know if this is even being investigated.  I say do whatever works.  If you have to wait a bit until after the stream is no longer live, so be it.  If you can get the livestream while it's live, so be it.  It seems to be very much a trial & error sort of thing.

**Comment by Fritz66** [comment]
ZDF changed their website which makes it difficult to download movies in .mp4 format (I don't like .webm).

I figured out how to download movies from the ZDF website.
1) Go to the movie's website like https://www.zdf.de/filme/freunde-mit-gewissen-vorzuegen-movie-100
2) Press CTRL+U to view the source code
3) Search for .m3u8, in this case it's https://zdfvod.akamaized.net/i/mp4/de/zdf/21/12/211220_sony_0105_sendung_nsf/1/211220_sony_0105_sendung_nsf,_508k_p9,_808k_p11,_1628k_p13,_3328k_p15,v15.mp4.csmil/master.m3u8
4) Download the movie with yt-dlp.
<img width="1343" height="359" alt="Bildschirmfoto zu 2025-08-07 18-12-48" src="https://github.com/user-attachments/assets/83452b52-a1d5-421c-9c7c-3fb50a7b8983" />

If the movie is offered in mutiple languages the audio track will be German if the URL always ends with .m3u8, which is how it should be.

Would be nice if Paul could update Video DownloadHelper to download movies on ZDF without using yt-dlp. Thanks!

**Reply by Mattmjs** [reply]
I don't think this is an issue, VDH is detecting what it finds and that is webm. The site isn't offering mp4.
The method you used didn't work, it doesn't find a m3u8 link when I tried it. I don't see this being fixed.

Webm files work fine on the computer, I have found that they don't play on the TV. Maybe newer TV's handle it.

**Comment by Fritz66** [comment]
If you can't read my post properly, then you shouldn't reply to it.
The method I use works perfectly for me, as described.

Also, my post is addressed to Paul.

**Reply by Mattmjs** [reply]
I did read it properly and followed the steps but for me it says nothing is found. But if works for you just continue with it because
I don't see Paul fixing anything regarding this site as there is already detection.
